generator/src/main/java/com/algorand/sdkutils/listeners/GoGenerator.java

```diff
@@ -303,13 +314,32 @@ private void addPathParameter(TypeDef type) {
         pathParameters.put(propName, gotype);
 
         // client functions
-        if (pathParameters.size() > 1) {
+        if (pathParameters.size() + bodyParameter.size() > 1) {
+            clientFunction.append(", ");
+        }
+        clientFunction.append(propName + " " + gotype);
+    }
+
+    private void addBodyParameter(TypeDef type) {
+        String gotype = goType(type.rawTypeName, type.isOfType("array"));
+        String propName = Tools.getCamelCase(
+                type.goPropertyName.isEmpty() ? type.propertyName : type.goPropertyName, 
+                        false);
+        bodyParameter.put(propName, gotype);
+
+        // client functions
+        if (pathParameters.size() + bodyParameter.size() > 1) {
             clientFunction.append(", ");
         }
         clientFunction.append(propName + " " + gotype);
     }
 
     private void addQueryParameter(TypeDef type) {
+        // Do not expose format property
+        if (type.javaTypeName.equals("Enums.Format")) {
```

Comment:
not sure I understand how is this decision made

Reply:
We are not exposing this parameter to the SDK. 
We silently set it to the default encoding.

Reply:
The SDKs have "real" transaction objects, so we opted to de-serialze the messagepack objects into those when possible.